src/mappings/entities/mapConfidentialAssetTransaction.ts

```diff
@@ -0,0 +1,84 @@
+import { Codec } from '@polkadot/types/types';
+import {
+  EventIdEnum,
+  ModuleIdEnum,
+  ConfidentialAssetHolder,
+  ConfidentialAssetTransaction,
+} from '../../types';
+import { getTextValue } from '../util';
+import { HandlerArgs } from './common';
+
+type AssetHolderParams = {
+  id: string;
+  accountId: string;
+  assetId: string;
+  amount: string;
+  balance: string;
+};
+
+const extractAssetHolderParams = (params: Codec[]): AssetHolderParams => {
+  const accountId = getTextValue(params[0]);
+  const assetId = getTextValue(params[1]);
+  const amount = getTextValue(params[2]);
+  const balance = getTextValue(params[3]);
+  const id = `${assetId}/${accountId}`;
+
+  return { id, accountId, assetId, amount, balance };
+};
+
+const handleConfidentialTransaction = async ({
+  blockId,
+  eventIdx,
+  eventId,
+  params,
+  block,
+}: HandlerArgs): Promise<void> => {
+  const { id, accountId, assetId, amount, balance } = extractAssetHolderParams(params);
+
+  const existingAssetHolder = await ConfidentialAssetHolder.get(id);
+
+  if (existingAssetHolder) {
+    existingAssetHolder.amount = balance;
+    existingAssetHolder.updatedBlockId = blockId;
+
+    await existingAssetHolder.save();
+  } else {
+    await ConfidentialAssetHolder.create({
+      id: `${assetId}/${accountId}`,
+      accountId,
+      assetId,
+      amount,
+      createdBlockId: blockId,
+      updatedBlockId: blockId,
+    }).save();
+  }
+
+  await ConfidentialAssetTransaction.create({
+    id: `${assetId}/${accountId}/${eventIdx}`,
+    assetId,
+    amount,
+    eventId,
+    eventIdx,
+    datetime: block.timestamp,
+    createdBlockId: blockId,
+    updatedBlockId: blockId,
+  }).save();
+};
+
+export const mapConfidentialAssetTransaction = async (args: HandlerArgs): Promise<void> => {
+  const { moduleId, eventId } = args;
+
+  if (moduleId !== ModuleIdEnum.confidentialasset) {
+    return;
+  }
+
+  if (
+    [
+      EventIdEnum.AccountDeposit,
+      EventIdEnum.AccountWithdraw,
+      EventIdEnum.AccountDepositIncoming,
```

Comment:
The way incoming is handled is different. Incoming is just a sign that amount will be deposited. It actually doesn't mean that amount is added. It would be great if you could confirm the working with Robert